spec/classes/datadog_agent_spec.rb

```diff
@@ -1909,6 +1909,62 @@
                 )
               }
             end
+
+            context 'with apm_enabled set to true and apm_obfuscation specified' do
```

Comment:
nit: we're missing a similar test for A5 to test the ini-style file generation (the conf footers), you might feel free to ignore, but just pointing it out.

Reply:
We don't pass this config to A5 at all, so there should be no change for A5. These settings end up in `$agent_config` which is only used when generating A6/7 config.

Reply:
Then why have changes to templates/datadog_apm_footer.conf.erb at all? I guess we can remove that, that's for A5 config.

Reply:
My bad, it is used by that template. I'll add a test.

Reply:
Actually the template is not working since the nested objects aren't output in `ini` format. I'm just going to remove it and not support this on A5, which I think is reasonable.